packages/core/typescript/TypeScriptFileUpdate.ts

```diff
@@ -65,64 +66,120 @@ export class TypeScriptFileUpdate {
 	 * @param linkPath Routing `path` to add
 	 * @param linkText Text of the route to add as `data.text`
 	 * @param routesVariable Name of the array variable holding routes
+	 * @param parentRoutePath __Optional__ If passed, will include the new route as a **child** of the specified route path
 	 */
-	public addRoute(filePath: string, linkPath: string, linkText: string, routesVariable = "routes") {
+	public addRoute(
+		filePath: string,
+		linkPath: string,
+		linkText: string,
+		routesVariable = DEFAULT_ROUTES_VARIABLE,
+		parentRoutePath?: string
+	) {
 		let className: string;
 		const fileSource = TsUtils.getFileSource(filePath);
 		const relativePath: string = Util.relativePath(this.targetPath, filePath, true, true);
-
+		routesVariable = routesVariable || DEFAULT_ROUTES_VARIABLE;
```

Comment:
There's literally no need to do this, that's the who point of default-initialized parameters - they assign the default value if none is passed. To skip over them you provide `undefined` to get the default. The proper call in the tests should be
`tsUpdate.addRoute("path/to/component", "child-1", "child one", undefined, "parent");` instead of `tsUpdate.addRoute("path/to/component", "child-1", "child one", "", "parent");`

That leads me to the two optional params at the end - frankly if you find the jumping over `routes` too annoying (it was really meant to be an the last one) - you can swap them and default the parent route to null as I think the route variable override is used even less, if at all.
Alternatively, can be split into two public facing methods that map back to this one (which would be hidden)

Reply:
For more on the params see https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters
Also if curious, check out this [playground](https://www.typescriptlang.org/play/index.html#code/GYVwdgxgLglg9mABAQwCaoPIBtUGUIAWccWAFAB4BciYIAtgEYCmATgDSICeiAvIgIwcAXr0QAmAJSIA3gFgAUIiWIICAM4kmAOixwA5qQAGVRABJp5AL6GJAbgXKV6zTv1HO1c52t2Hy1WAaWNq6BoZCntJCPvaKyixMUCAsSOSIANRcGYhCsZYKCgFBIW5omDj4RCSkAMwc4KhMwDBgTKiIbDUSvvJAA) and change the target to ES5 to see what the functionality polyfil does:
```js
// function addOldSchool(x, y = 1, z = 2) {
function addOldSchool(x, y, z) {
    if (y === void 0) { y = 1; }
    if (z === void 0) { z = 2; }
```